python/ray/tests/test_runtime_env_ray_minimal.py

```diff
@@ -103,5 +106,70 @@ def test_ray_client_init(call_ray_start):
     assert "ray[default]" in str(excinfo.value)
 
 
+@pytest.mark.skipif(
+    sys.platform == "win32", reason="runtime_env unsupported on Windows.")
+@pytest.mark.skipif(
+    os.environ.get("RAY_MINIMAL") != "1",
+    reason="This test is only run in CI with a minimal Ray installation.")
+@pytest.mark.parametrize("option", ["working_dir", "py_modules"])
+def test_remote_package_uri(start_cluster, option):
+    """
+    Test is based on test_remote_package_uri from
+    test_runtime_env_working_dir.py file. Ensures that runtime_env logic
+    gracefully raises error if smart_open is not installed and a remote URI
+    is specified for a dependency package.
+    """
+    cluster, address = start_cluster
+    uri = "s3://bucket/file.zip"
+
+    if option == "working_dir":
+        env = {"working_dir": uri}
+    elif option == "py_modules":
+        env = {"py_modules": [uri]}
+
+    try:
+        with pytest.raises(Exception) as err:
+            ray.init(address, runtime_env=env)
+        assert "pip install smart_open" in str(err.value)
+    except Failed:
```

Comment:
Why do we need this try-except?  The underscore in front of `_pytest` makes me think it's some internal module we shouldn't need to use, but I haven't seen it before so I'm not sure

Reply:
+1, shouldn't be using internal types like this

Reply:
I'm not actually sure what this except clause is doing -- do we always expect it to not raise an exception?